docs/collecting-data/collecting-from-own-applications/python-tracker/tracking-specific-events/index.md

```diff
@@ -1,528 +1,282 @@
-| `category` | Item's category | No | String |
-| `unit_price` | Item's price | No | Int or Float |
-| `currency` | Currency | No | String |
-| `context` | Custom context for the event | No | List(SelfDescribingJson) |
-| `tstamp` | When the pageview occurred | No | Positive integer |
-| `event_subject` (since v0.9.0) | The subject for the event | No | Subject |
-
-Basic example:
+from snowplow_tracker import SelfDescribingJson
 
-```python
-t.track_remove_from_cart("123", 1)
+geo_context = SelfDescribingJson(
+  "iglu:com.snowplowanalytics.snowplow/geolocation_context/jsonschema/1-0-0",
+  {
+    "latitude": -23.2,
+    "longitude": 43.0
+  }
+)
 ```
 
-Advanced example:
+As another example, if a visitor arrives on a page advertising a movie, the context object might look like this (`movie_poster` is custom context, not predefined):
 
 ```python
-t.track_remove_from_cart("123", 2, "The Devil's Dance", "Books", 23.99, "USD")
+poster_context = SelfDescribingJson(
+  "iglu:com.acme_company/movie_poster/jsonschema/2-1-1",
+  {
+    "movie_name": "Solaris",
+    "poster_country": "JP",
+    "poster_year": "1978-01-01"
+  }
+)
 ```
 
-### Track form change
-
-Use `track_from_change()` to track changes in website form inputs over session. Arguments are:
-
-| **Argument** | **Description** | **Required?** | ******Type****** |
-| --- | --- | --- | --- |
-| `form_id` | ID attribute of the HTML form | Yes | Non-empty string |
-| `element_id` | ID attribute of the HTML element | Yes | String |
-| `node_name` | Type of input element | Yes | [Valid node_name](https://github.com/snowplow/iglu-central/blob/53589a5dd41f2f88cabb359488019e0ebb72ec49/schemas/com.snowplowanalytics.snowplow/change_form/jsonschema/1-0-0#L20) |
-| `value` | Value of input element | Yes | String |
-| `type_` | Type of data the element represents | No | Non-empty string |
-| `element_classes` | Classes of the HTML element | No | List(string) |
-| `context` | Custom context for the event | No | List(SelfDescribingJson) |
-| `tstamp` | When the pageview occurred | No | Positive integer |
-| `event_subject` (since v0.9.0) | The subject for the event | No | Subject |
-
-Basic example:
+This is how to fire a page view event with both above contexts:
 
 ```python
-t.track_form_change("signupForm", "ageInput", "age", "24")
-```
-
-Advanced example:
+page_view = PageView(
+  page_url="https://www.snowplow.io",
+  page_title="Homepage",
+  context=[poster_context, geo_context]
+)
 
-```python
-t.track_form_change("signupForm", "ageInput", "age", "24", "number", ["signup__number", "form__red"])
+t.track(page_view)
 ```
 
-### Track submitted forms
+**Important:** Even if only one custom context is being attached to an event, it still needs to be wrapped in an list.
 
-Use `track_form_submit()` to track sumbitted forms. Arguments are:
+Note also that you should not pass in an empty list of contexts as this will fail validation. Instead of an empty list, you can pass in `None`.
```

Comment:
Is this really true? That sounds like something that should be handled in the tracker rather than bothering users about it – if the list of context is empty, ignore it.

Reply:
This is something we fixed in V1 so I'll remove this line!